api.md

```diff
@@ -919,6 +922,82 @@ The expected output:
 
 ---
 
+## interpolate() [](interpolate)
+
+The `interpolate` function does linear interpolation for missing values.
+It can only be used in an aggregation query with [time_bucket_gapfill](#time_bucket_gapfill).
+The `interpolate` function call cannot be nested inside other function calls.
+
+#### Required Arguments [](interpolate-required-arguments)
+
+|Name|Description|
+|---|---|
+| `value` | The value to interpolate (int2/int4/int8/float4/float8) |
+
+#### Optional Arguments [](interpolate-optional-arguments)
+
+|Name|Description|
+|---|---|
+| `prev` | The lookup expression for values before the gapfill time range (record) |
+| `next` | The lookup expression for values after the gapfill time range (record) |
+
+The `prev` and `next` expression will only be evaluated when no suitable value is returned by the outer query.
```

Comment:
Agree with @dianasaur323 that just something more is needed to clarify what prev/next are used for.  Something perhaps like (please review for correctness):

Because the interpolation function relies on having values before and after each bucketed period to compute the interpolated value, it can have difficulty in calculating the interpolation for the first and last time bucket if those buckets do not otherwise contain valid values.  For example, the interpolation would require looking *before* this first time bucket period, yet the query's outer time predicate `WHERE time > ...` normally restricts the function to only evaluate values *within* this time range.  Thus, the `prev` and `next` expression tell the function how to look for values *outside* of the range specified by the time predicate; these expressions will only be evaluated when no suitable value is returned by the outer query.

Reply:
@dianasaur323 its almost true, there is no valid value constraint, explicit NULLS returned from outer query won't be silently ignored, if you dont want them you need to exclude them in WHERE clause